docs/solidity-by-example.rst

```diff
@@ -620,4 +620,52 @@ Safe Remote Purchase
 Micropayment Channel
 ********************
 
-To be written.
+::
+
+    pragma solidity ^0.4.11;
+    contract Channel {
+        address[2] users;
+        uint deposit;
+        uint timeout;
+        uint closeTimestamp;
+        uint closeSequenceNumber;
+        uint closeBalanceUser0;
+        enum State { Opened, Accepted, Closing, Closed }
+        State state;
+        function Channel(address _other, uint _timeout) payable {
+            users[0] = msg.sender;
+            users[1] = _other;
+            deposit = msg.value;
+            require(deposit < 2**250);
+            require(_timeout < 2 days);
+            timeout = _timeout;
+        }
+        function accept() payable {
+            require(msg.sender == users[1]);
+            require(msg.value == deposit);
+            require(state == State.Opened);
+            state = State.Accepted;
+        }
+        function requestClose(uint _balanceUser0, uint _sequenceNumber, uint8 v, bytes32 r, bytes32 s) {
+            require(state == State.Accepted || state == State.Closing);
```

Comment:
Why allow closing multiple times? Each time the timestamp/balance/sequence is updated.

Reply:
@chriseth this is still an open question